devito/types/sparse.py

```diff
@@ -95,12 +95,14 @@ def __shape_setup__(cls, **kwargs):
         # A Grid must have been provided
         if grid is None:
             raise TypeError('Need `grid` argument')
-        shape = kwargs.get('shape')
+        shape = kwargs.get('shape', kwargs.get('shape_global'))
         dimensions = kwargs.get('dimensions')
         npoint = kwargs.get('npoint', kwargs.get('npoint_global'))
         glb_npoint = SparseDistributor.decompose(npoint, grid.distributor)
         if shape is None:
             loc_shape = (glb_npoint[grid.distributor.myrank],)
+        elif dimensions is None:
```

Comment:
I don't understand how this is possible and, even then, what it would mean to assign `loc_shape = shape`. If it's an obscure case could you add a short comment?

Reply:
but why would one rebuild w/o dimensions?